Google.Api.Gax.Grpc/Rest/HttpRulePathPattern.cs

```diff
@@ -103,15 +109,30 @@ internal sealed class HttpRulePathPatternSegment
             /// Creates a new path pattern segment from a path segment with the bound field, e.g. `{resource.id}` in `v1/resources/{resource.id}`
             /// </summary>
             /// <param name="propertyPath">A </param>
+            /// <param name="boundFieldPattern"></param>
             /// <param name="descriptor"></param>
             /// <returns></returns>
-            internal static HttpRulePathPatternSegment CreateFromBoundField(string propertyPath, MessageDescriptor descriptor)
+            internal static HttpRulePathPatternSegment CreateFromBoundField(string propertyPath,
+                string boundFieldPattern, MessageDescriptor descriptor)
             {
                 int periodIndex = propertyPath.IndexOf('.');
                 bool singleFieldPath = periodIndex == -1;
                 string fieldName = singleFieldPath  ? propertyPath : propertyPath.Substring(0, periodIndex);
 
-                return new HttpRulePathPatternSegment(fieldName, CreatePropertyAccessor(descriptor, propertyPath));
+                var propertyAccessor = CreatePropertyAccessor(descriptor, propertyPath);
+
+                // Per `google/api/http.proto`, if the pattern has  multiple path segments, such as `"{var=foo/*}"`,
+                // the `/` symbol should not be encoded, otherwise it should.
+                //TODO: [virost, 2021-12] Match pattern on the bound field, in addition to selecting the escape function
+                Func<string, string> escape_regular = value => Uri.EscapeDataString(value);
```

Comment:
Nit: please rename to `escapeRegular` and `escapeMultisegment`

Reply:
👍